pxf/pxf-api/src/main/java/org/apache/hawq/pxf/api/FilterParser.java

```diff
@@ -396,4 +387,12 @@ private Operation reverseOp(Operation operation) {
         operatorTranslationMap.put(8, Operation.HDOP_LIKE);
         return operatorTranslationMap;
     }
+
+    static private Map<Integer, LogicalOperation> initLogicalOperatorTransMap() {
+        Map<Integer, LogicalOperation> integerLogicalOperationMap = new HashMap<>();
```

Comment:
please review http://stackoverflow.com/questions/5292790/convert-integer-value-to-matching-java-enum for discussion on dangers of relying on ordinal values. I think for us, we might want to maintain explicit number as the private member of enum, then build and cache the array once for all future access.